File: Arboles/graphs/ejemplo-matriz-ady.py
from sys import stdin

def main():
    adj = [[0 for i in range(10)] for i in range(10)]
    n = int(stdin.readline())

    for i in range(n):
        vals = list(map(int, stdin.readline().split()))
        u = vals[0]
        for j in range(u):
            v = vals[j + 1]
            adj[i][v] = 1

    print("Grafo")
    for i in range(n):
        for j in range(n):
            print("%d" % adj[i][j], end = ' ')
        print("")

main()

File: Arboles/graphs/ejemplo-matriz-ady.cpp
#include <vector>
#include <map>
#include <string>
#include <cstdio>
#include <iostream>

using namespace std;

int main(){
  int n, m, u, v, i, j;
  int adj[10][10] = {0};

  cin >> n;

  for(i = 0; i < n; ++i){
    cin >> u;
    for(j = 0; j < u; ++j){
      cin >> v;
      adj[i][v] = 1;
    }
  }

  cout << "Grafo" << endl;

  for(i = 0; i < n; ++i){
    for(j = 0; j < n; ++j)
      cout << adj[i][j] << " ";
    cout << endl;
  }

  return 0;
}
